# Honors problems

***
*Problem 114*:  Counting block combinations I

In [2]:
m = 3
n = 50
ways = [0]*(n+1)
ways[0] = ways[1] = ways[2] = 1
for i in range(m, n+1):
    Sum = 0
    for j in range(0,i-m):
        Sum += ways[j]
    ways[i] = ways[i-1] + Sum + 1
print(ways[n])

16475640049


The number of ways of filling a row of length 50 is **16475640049**

***
*Problem 115*: Counting block combinations II